crates/net/network/src/frost/messages.rs

```
@@ -494,29 +491,32 @@ impl FrostProtoMessage {
                 ))
             }
             FrostProtoMessageId::WalletState => {
-                // utxos
-                let utxos_len = u64::from_le_bytes(buf[..8].try_into().unwrap()) as usize;
-                buf.advance(8);
-                let utxos = buf[..utxos_len].to_vec();
-                buf.advance(utxos_len);
-
-                // tracked txs
-                let tracked_txs_len = u64::from_le_bytes(buf[..8].try_into().unwrap()) as usize;
-                buf.advance(8);
-                let tracked_txs = buf[..tracked_txs_len].to_vec();
-                buf.advance(tracked_txs_len);
-
-                // pending pegouts
-                let pending_pegouts_len = u64::from_le_bytes(buf[..8].try_into().unwrap()) as usize;
-                buf.advance(8);
-                let pending_pegouts = buf[..pending_pegouts_len].to_vec();
-                buf.advance(pending_pegouts_len);
-
-                FrostProtoMessageKind::WalletState(WalletStateRequest::new(
-                    utxos,
-                    tracked_txs,
-                    pending_pegouts,
-                ))
+                // Decode uuid
+                let uuid_len = u32::from_le_bytes(
+                    buf[..4].try_into().expect("Buffer underflow for uuid length"),
+                ) as usize;
+                buf.advance(4);
+                let uuid =
+                    String::from_utf8(buf[..uuid_len].to_vec()).expect("Invalid UTF-8 in uuid");
```

Non-utf8 characters seems like something we should return an error for. Especially when we are deserializing.

---

There wont be any non utf-8 chars. uuid produces a fully compatible utf-8 string.

---

Also, in all these ser/des methods we expect or unwrap. We might need a ticket to refactor all of this, but this is a separate concern.

---

>There wont be any non utf-8 chars. uuid produces a fully compatible utf-8 string.

Does this mean that if a peer provides me an non uuid as an id that it would cause a panic? 

>>Also, in all these ser/des methods we expect or unwrap. We might need a ticket to refactor all of this, but this is a separate concern.

Fair enough. If the answer to the above is yes i do think it should be addressed.

---

@0xBEEFCAF3 I have resolved all unwraps (this was an audit finding) and now all errors are being handled properly. Please check and resolve.